src/Capability/Discovery/DiscoveryState.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/*
+ * This file is part of the official PHP MCP SDK.
+ *
+ * A collaboration between Symfony and the PHP Foundation.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Mcp\Capability\Discovery;
+
+use Mcp\Capability\Registry\PromptReference;
+use Mcp\Capability\Registry\ResourceReference;
+use Mcp\Capability\Registry\ResourceTemplateReference;
+use Mcp\Capability\Registry\ToolReference;
+
+/**
+ * Represents the state of discovered MCP capabilities.
+ *
+ * This class encapsulates all discovered elements (tools, resources, prompts, resource templates)
+ * and provides methods to apply this state to a registry.
+ *
+ * @author Xentixar <xentixar@gmail.com>
+ */
+class DiscoveryState
```

Comment:
can be final, right? even without interface
```suggestion
final class DiscoveryState
```